### PR TITLE
RD-4194 Added ability to enter non-existing secret to the Terraform modal

### PR DESCRIPTION
**Pull request by NorbertAmartus**
## Description

Changes consists of adding the ability for the user to enter non-existing secret

## Screenshots / Videos

N/A

## Checklist

- [x] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires a documentation update.
- [x] I added proper labels to this PR.

## Tests

 

[System tests](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/1995) has been executed (only in context of the `TerraformModal`).

## Documentation

Documentation stated that the user can select only secrets from those already created.
As we are starting to support ability to use non-existing secrets, I've updated the documentation correspondingly, within [this PR](https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2095)

After merging the [documentation PR](https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2095) I'll add another commit here, to synchronise documentation changes 🚀

### CODE REVIEW

File: test/cypress/integration/widgets/blueprints_spec.ts [hunk 1]
```diff
@@ -463,14 +463,22 @@ describe('Blueprints widget', () => {
             cy.contains('.item', source).click();
         }
 
+        function initializeFirstSegmentRow(segmentName: string) {
```
Comment (kubama) [inline test/cypress/integration/widgets/blueprints_spec.ts:466]: How about:
```suggestion
        function addFirstSegmentRow(segmentName: string) {
```
The term 'initialize' is not very common in terms of UI controls.

File: test/cypress/integration/widgets/blueprints_spec.ts [hunk 2]
```diff
@@ -558,6 +567,32 @@ describe('Blueprints widget', () => {
             });
         });
 
+        it('enable to enter non-existing secret value', () => {
```
Comment (kubama) [inline test/cypress/integration/widgets/blueprints_spec.ts:570]: Shouldn't it be:
```suggestion
        it('enable to enter non-existing secret key', () => {
```
?

File: test/cypress/integration/widgets/blueprints_spec.ts [hunk 3]
```diff
@@ -558,6 +567,32 @@ describe('Blueprints widget', () => {
...
+            const validVariableName = 'abc';
+            const notExistingSecretKey = `${blueprintNamePrefix}_terraform_secret`;
+
+            const setNotExistingSecretValue = () => {
```
Comment (kubama) [inline test/cypress/integration/widgets/blueprints_spec.ts:574]:
```suggestion
            const setNotExistingSecretKey = () => {
```

File: test/cypress/integration/widgets/blueprints_spec.ts (hunk 2 above)
Comment (NorbertAmartus) [reply test/cypress/integration/widgets/blueprints_spec.ts:570]: Oh yeah, I've got myself confused about that one
Thanks for pointing that out! :)

File: test/cypress/integration/widgets/blueprints_spec.ts (hunk 1 above)
Comment (NorbertAmartus) [reply test/cypress/integration/widgets/blueprints_spec.ts:466]: Sure, in that case I'll modify that accordingly 😉

File: test/cypress/integration/widgets/blueprints_spec.ts (hunk 3 above)
Comment (NorbertAmartus) [reply test/cypress/integration/widgets/blueprints_spec.ts]: I was wondering if the phrase "Value" or "Key" is even necessary here, as in the modal is being shown as just a "Secret"
I've modified names to drop od those phrases, to be more consistent with the displayed UI

Let me know what do you think about that 😉

**Review by kubama — APPROVED** [review]

**Comment by NorbertAmartus** [comment]
Synchronised the documentation 😁

**Review by kubama — APPROVED** [review]

**Pull request merged**